registry/remote/repository.go

```diff
@@ -1463,14 +1468,15 @@ func (s *manifestStore) updateReferrersIndex(ctx context.Context, subject ocispe
 			}
 		}
 
-		// 4. delete the dangling original referrers index
-		if !skipDelete {
-			if err := s.repo.delete(ctx, oldIndexDesc, true); err != nil {
-				return &ReferrersError{
-					Op:      opDeleteReferrersIndex,
-					Err:     fmt.Errorf("failed to delete dangling referrers index %s for referrers tag %s: %w", oldIndexDesc.Digest.String(), referrersTag, err),
-					Subject: subject,
-				}
+		// 4. delete the dangling original referrers index, if applicable
+		if s.repo.SkipReferrersGC || noOldIndex {
```

Comment:
Shouldn't `noOldIndex` already be implied by `len(oldReferrers) == 0`?

Reply:
No. Consider the following cases:
1. Old referrers index exists and contains at least one referrers: `noOldIndex == false`, `len(oldReferrers) > 0`
2. Old referrers index exists but contains no referrer: `noOldIndex == false`, `len(oldReferrers) == 0`
3. Old referrers index does not exist (server returns 404):  `noOldIndex == true`, `len(oldReferrers) == 0`

We skip deleting the old index when `skipReferrersGC == true` or the old index is not present (case 3).

Reply:
How about `s.repo.SkipReferrersGC || content.Equal(oldIndexDesc, ocispec.Descriptor{})`

Reply:
The main purpose of this comment is: whether an old index exists is already implied by returning value of function `referrersFromIndex`.

Reply:
The suggested way does have the same effect as the current code. But I think having `noOldIndex` is more readable. Otherwise we will need extra comment to explain what does `content.Equal(oldIndexDesc, ocispec.Descriptor{})` mean.

Reply:
Refactored a little bit.